tests/builder/help.rs

```diff
@@ -3967,3 +3967,79 @@ Print this message or the help of the given subcommand(s)
 "#]];
     utils::assert_output(cmd, "parent -h", expected, false);
 }
+
+#[test]
+fn mixed_argument_types() {
```

Comment:
Could you re-orgaize this PR where this commit as added in its own commit but changed so its passing, then this commit would change the test to pass with the new logic.  The diff between these two commits would then show how the behavior changed.

See https://github.com/clap-rs/clap/blob/master/CONTRIBUTING.md#preparing-the-pr for more details